### ISSUE TITLE
Escape key closes site preview pane too aggressively

### ISSUE THREAD

**Issue by jeryj** [Automattic/wp-calypso#98236]
### Quick summary

The escape key is used to close the site preview pane, even if you are using Escape to close a popover. Escape to close the Site Management/Preview Pane was proposed in https://github.com/Automattic/dotcom-forge/issues/6976 and added in https://github.com/Automattic/wp-calypso/pull/90422. 

I see two possibilities:

1. Remove Escape to close the preview panel
Escape makes sense to close modals and popovers, but it's unconventional to reroute an entire page on Escape. I'm not sure what the rationale was for adding this behavior, so I'd be interested to hear more here.

2. Keep Escape and fix the following:

- [ ] Escape from within popovers and things that should handle Escape (three dot menus, etc) should not also close the site preview pane
- [ ] Pressing Escape should only go back when the event comes from inside the site preview pane, not from anywhere on the page
- [ ] Focus should be handled appropriately. If the intention is to move between sites more easily, focus should be moved to the row/site title of the closed site preview. Right now it is not handled, so nothing gets focused when the panel is closed.

My initial take is to go with route 1 -- remove the Escape to close the panel. I don't see strong rationale for adding it. However, this is a first glance at it, and I know I'm missing a lot of history and context here.

### Steps to reproduce

1. http://calypso.localhost:3000/sites
2. Click into a site
3. Open a triple dot popover
4. Press escape to close the popover 

https://github.com/user-attachments/assets/55fd05ad-f880-4a10-aafe-ce7ff9b6dacb

### What you expected to happen

Popover is closed and focus returned to the button that opened it

### What actually happened

Bug: Preview site panel is also closed
Bug: Focus is not handled
Bug: Escape should not be handled when focus is outside the preview panel

### Impact

All

### Available workarounds?

No but the platform is still usable

**Comment by Robertght** [comment]
📌 REPRODUCTION RESULTS
- Tested on the platform - Replicated

📌 ACTIONS
- Triaged
- Lowered priority to Normal
